superset/assets/javascripts/modules/superset.js

```diff
@@ -215,13 +237,22 @@ const px = function (state) {
         container.fadeTo(0.5, 0.25);
         sliceCell.addClass('slice-cell-highlight');
         container.css('height', this.height());
+        const asyncAnnotations = (formData.annotation_layers || [])
+          .map(x => runAnnotationQuery(x, timeout, formData));
         $.ajax({
           url: this.jsonEndpoint(formDataExtra),
           timeout: timeout * 1000,
           success: (queryResponse) => {
             try {
-              vizMap[formData.viz_type](this, queryResponse);
               this.done(queryResponse);
+              // render when all the annotations are available
+              Promise.all(asyncAnnotations)
+                .then((annotations) => {
+                  this.annotationData = annotations
+                    .reduce((data, a) => ({ ...data, ...a }), {});
+                  return Promise.resolve();
+                })
+                .then(() => vizMap[formData.viz_type](this, queryResponse));
```

Comment:
do you need a `.catch` in this promise chain? again ideally (in a dashboard) it'd be ideal to see any errors that occur as opposed to them being only in the explorer controls.

Reply:
This has been addressed

Reply:
Shouldn't matter as this file should not exist anymore...